### PR TITLE
More consensus unification

### PR DESCRIPTION
**Pull request by olgavrou**
Continuing issue #300 

- some renaming/reshuffling
- merged consensus_msg_raft and consensus_msg_pbft into consensus_msg
- frontend to use consensus instead of raft

### CODE REVIEW

**Comment by cimetrics** [comment]
Performance report: 
 images 
 **Have a look before merging** :bar_chart: :eyes: :thumbsup:

File: src/consensus/ledgerenclavetypes.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/ringbuffer_types.h"
+
+namespace consensus
```
Comment (jumaffre) [inline src/consensus/ledgerenclavetypes.h]: Thinking about it, I am not sure this should be under `consensus`. It is used to fetch entries from the ledger (e.g. `nodestate.h`) in places that are not related to the consensus. Could we move it to `node/`?

File: src/consensus/ledgerenclavetypes.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+  using Index = uint64_t;
+  /// Consensus-related ringbuffer messages
+  enum : ringbuffer::Message
+  {
+    /// Request individual log entries. Enclave -> Host
+    DEFINE_RINGBUFFER_MSG_TYPE(log_get),
```
Comment (jumaffre) [inline src/consensus/ledgerenclavetypes.h]: Since this is a refactoring/renaming PR and if it's not too tedious, could we rename these from `log_` to `ledger_`?

Comment (olgavrou) [reply src/consensus/ledgerenclavetypes.h]: ah good catch

File: src/consensus/ledgerenclave.h [hunk 3]
```diff
@@ -25,18 +25,18 @@ namespace raft
...
-     * Put a single entry to be written the ledger, when leader.
+     * Put a single entry to be written the ledger, when primary.
      *
      * @param entry Serialised entry
      */
     void put_entry(const std::vector<uint8_t>& entry)
     {
       // write the message
-      RINGBUFFER_WRITE_MESSAGE(raft::log_append, to_host, entry);
+      RINGBUFFER_WRITE_MESSAGE(consensus::log_append, to_host, entry);
```
Comment (achamayou) [inline src/consensus/ledgerenclave.h]: ledger_append

File: src/consensus/ledgerenclavetypes.h (hunk 1 above)
Comment (olgavrou) [reply src/consensus/ledgerenclavetypes.h]: I am not 100% convinced because they are largely used by the consensus to write/read from the ledger, even if it used by the nodestate.h, and it makes it clear that the consensus is the one that writes to the ledger

File: src/node/nodestate.h [hunk 4]
```diff
@@ -1250,14 +1246,14 @@ namespace ccf
...
 
-    void read_ledger_idx(raft::Index idx)
+    void read_ledger_idx(consensus::Index idx)
     {
-      RINGBUFFER_WRITE_MESSAGE(raft::log_get, to_host, idx);
+      RINGBUFFER_WRITE_MESSAGE(consensus::log_get, to_host, idx);
     }
 
-    void log_truncate(raft::Index idx)
+    void log_truncate(consensus::Index idx)
```
Comment (eddyashton) [inline src/node/nodestate.h]: While renaming the messages to `log_`, this method should also be `ledger_truncate`

File: src/consensus/ledgerenclavetypes.h (hunk 2 above)
Comment (olgavrou) [reply src/consensus/ledgerenclavetypes.h]: done

File: src/node/nodestate.h (hunk 4 above)
Comment (olgavrou) [reply src/node/nodestate.h]: done

**Comment by cimetrics** [comment]
images
